### PR TITLE
Making Offer-up Work with GHCS

### PR DESCRIPTION
**Pull request by amessbee**
Changes include:
- script to make ports visible
- `.devcontainer` directory and file
- addition of env variables for domain and codespace name
- fix RPC and API ENDPOINTS in UI for wallet connection
- instructions addition in README

### CODE REVIEW

**Comment by amessbee** [comment]
@rabi-siddique can you please review this. @mujahidkay has already tested it, and I think we can merge it after you get to test it and find it reasonably stable. I added a few instructions in main README.md apart from other GHCS related changes. Thanks.

File: ui/src/App.tsx [hunk 1]
```diff
@@ -25,6 +25,16 @@ const ENDPOINTS = {
...
+const codeSpaceHostName = import.meta.env.VITE_HOSTNAME;
+
+const codeSpaceDomain = import.meta.env
+  .VITE_GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN;
+
+if (codeSpaceHostName) {
+  ENDPOINTS.API = `https://${codeSpaceHostName}-1317.${codeSpaceDomain}`;
+  ENDPOINTS.RPC = `https://${codeSpaceHostName}-26657.${codeSpaceDomain}`;
+}
+
```
Comment (rabi-siddique) [inline ui/src/App.tsx:37]:
```suggestion
if (codeSpaceHostName && codeSpaceDomain) {
ENDPOINTS.API = `https://${codeSpaceHostName}-1317.${codeSpaceDomain}`;
ENDPOINTS.RPC = `https://${codeSpaceHostName}-26657.${codeSpaceDomain}`;
} else {
console.error("Missing environment variables: VITE_HOSTNAME or VITE_GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN");
}
```

**Comment by rabi-siddique** [comment]
Getting CORS error while connecting to the wallet when using codespace:
Screenshot 2024-08-16 at 18 42 46

**Comment by rabi-siddique** [comment]
> Getting CORS error while connecting to the wallet when using codespace: image

My bad. I did not make the ports public. After making the ports public, I was able to connect the wallet and make the offer. There was an element of flakiness too but I am assuming it was due to my Internet. Will give it another try in a while. 

Looks good overall :)

**Review by rabi-siddique — APPROVED** [review]

**Pull request merged**